stdlib/string.ml

```diff
@@ -100,15 +100,17 @@ let trim s =
     then bts (B.trim (bos s))
   else s
 
-let escaped s =
-  let rec needs_escape i =
-    if i >= length s then false else
-      match unsafe_get s i with
-      | '\"' | '\\' | '\n' | '\t' | '\r' | '\b' -> true
-      | ' ' .. '~' -> needs_escape (i+1)
-      | _ -> true
+let needs_escape s =
+  let rec aux s n i =
+    if i >= n then false else
+      match String.unsafe_get s i with
+      | '\"' | '\\' | '\000'..'\031' | '\127'.. '\255' -> true
```

Comment:
Side note: it is "tempting" to drop the check on the length altogether (also avoiding the need to compute the length), relying on the fact that `String.unsafe_get s (String.length s)` returns `\000`.  The trouble, in addition to stressing @xavierleroy, is that this doesn't work with js_of_ocaml (and presumably Bucklescript).